examples/coappearances/coappearances.cpp

```diff
@@ -212,6 +212,108 @@ convert( const char* json_path, const char* archive_path )
     builder.set_strings( {strings.data( ), strings.size( )} );
 }
 
+// Note: The linear search below is slow for big graphs for obvious reasons, where it is much better
+// to build a lookup table. However, for our small graph this works fine.
+std::vector< uint32_t >
+get_neighbors( co::Graph graph, uint32_t vertex_ref )
+{
+    std::vector< uint32_t > neighbors;
+    for ( auto e : graph.edges( ) )
+    {
+        if ( e.a_ref == vertex_ref )
+        {
+            neighbors.push_back( e.b_ref );
+        }
+        else if ( e.b_ref == vertex_ref )
+        {
+            neighbors.push_back( e.a_ref );
+        }
+    }
+    return neighbors;
+}
+
+size_t
+calculate_num_connected_components( co::Graph graph )
+{
+    size_t num_connected_components = 0;
+
+    std::vector< bool > visited( graph.vertices( ).size( ) );
+    uint32_t num_visited = 0;
+    std::vector< uint32_t > stack;
+
+    while ( num_visited != graph.vertices( ).size( ) )
+    {
+        stack.clear( );
+        // find first non-visited vertex
+        for ( uint32_t vertex_ref = 0; vertex_ref < graph.vertices( ).size( ); ++vertex_ref )
+        {
+            if ( !visited[ vertex_ref ] )
+            {
+                stack.push_back( vertex_ref );
+                break;
+            }
+        }
+        // depth first search
+        while ( !stack.empty( ) )
+        {
+            uint32_t vertex_ref = stack.back( );
+            stack.pop_back( );
+
+            if ( visited[ vertex_ref ] )
+            {
+                continue;
+            }
+
+            visited[ vertex_ref ] = true;
+            num_visited++;
+
+            for ( auto neighbor_ref : get_neighbors( graph, vertex_ref ) )
+            {
+                if ( !visited[ neighbor_ref ] )
+                {
+                    stack.push_back( neighbor_ref );
+                }
+            }
+        }
+        num_connected_components++;
+    }
+    return num_connected_components;
+}
+
+void
+calculate( const char* archive_path )
+{
+    auto graph = co::Graph::open( flatdata::FileResourceStorage::create( archive_path ) );
+    if ( !graph )
+    {
+        throw std::runtime_error( std::string( "Could not open graph at: " ) + archive_path );
+    }
+
+    std::string subarchive_path = std::string( archive_path ) + "/calculated_data";
```

Comment:
No need to do this, just open a builder for the main archive, and ask for the builder of the subarchive